core/crypto/nike/nike.go

```diff
@@ -46,18 +54,21 @@ type PublicKey interface {
 	Blind(blindingFactor []byte) error
```

Comment:
This here should also just take a `*PrivateKey`

Reply:
When using the Blind method to implement Sphinx, we have a raw byte slice, not a private key!
Here is where we derive the blinding factors:

https://github.com/katzenpost/katzenpost/blob/main/core/sphinx/internal/crypto/crypto.go#L181-L182

This code has been like this for years.
You can't block this PR because of this.
If you want me to change it... I can do that in another pull request.